### PR TITLE
Mat 5381 refactor human readable generation

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5381](https://jira.cms.gov/browse/MAT-5381)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Comment by sb-cecilialiu** [comment]
Special thanks to Ashok for his great help!

**Comment by sb-cecilialiu** [comment]
Special thanks to Ashok for his great help!!

File: pom.xml [hunk 1]
```diff
@@ -133,6 +135,28 @@
...
+		<dependency>
+	  		<groupId>ca.uhn.hapi.fhir</groupId>
+	  		<artifactId>hapi-fhir-structures-r5</artifactId>
+	  		<version>${hapi.fhir.r5.version}</version>
+		</dependency> 
+		<dependency>
+	  		<groupId>ca.uhn.hapi.fhir</groupId>
+	  		<artifactId>hapi-fhir-validation</artifactId>
+	  		<version>${hapi.fhir.r5.version}</version>
+		</dependency>
```
Comment (adongare) [inline pom.xml]: does this need to be added? I thought it was part of some other package.

File: src/main/java/gov/cms/madie/madiefhirservice/exceptions/HumanReadableInvalidException.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package gov.cms.madie.madiefhirservice.exceptions;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.UNPROCESSABLE_ENTITY)
+@Slf4j
+public class HumanReadableInvalidException extends RuntimeException {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/exceptions/HumanReadableInvalidException.java]: did not find usage for this exception. maybe unused?

File: pom.xml (hunk 1 above)
Comment (sb-cecilialiu) [reply pom.xml]: Thanks Ashok! Removed those.

File: src/main/java/gov/cms/madie/madiefhirservice/exceptions/HumanReadableInvalidException.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/exceptions/HumanReadableInvalidException.java]: Removed as well, thanks!

**Review by adongare — DISMISSED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  private final ElmTranslatorClient elmTranslatorClient;
+
+  private static final String EFFECTIVE_DATA_REQUIREMENT_URL =
+      "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-effectiveDataRequirements";
+
+  public String generateHumanReadable(
+      Measure madieMeasure, String accessToken, Bundle bundleResource) {
+
+    if (bundleResource != null) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: Minor suggestion: Invert this `if` and throw the appropriate exception. It reduces the number nested `if`'s and makes it easier to link the exception to the condition.

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      }
+      try {
+        Optional<Bundle.BundleEntryComponent> measureEntry = getMeasureEntry(bundleResource);
+        if (measureEntry.isEmpty()) {
+          log.error("Unable to find measure entry for measure {}", madieMeasure.getId());
+          throw new ResourceNotFoundException("measure entry", madieMeasure.getId());
+        }
+        Resource measureResource = measureEntry.get().getResource();
+
+        // converting measure resource from R4 to R5
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: If possible, please include why we need to convert from R4 to R5 in this comment.

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  private Optional<Bundle.BundleEntryComponent> getMeasureEntry(Bundle bundleResource) {
+    return bundleResource.getEntry().stream()
+        .filter(
+            entry ->
+                StringUtils.equalsIgnoreCase(
+                    "Measure", entry.getResource().getResourceType().toString()))
+        .findFirst();
+  }
+
+  private Extension createExtension() {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]:
```suggestion
  private Extension createEffectiveDataRequirementExtension() {
```
More detailed name to avoid accidentally using it in some other HR section.

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java (hunk 4 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: @jkotanchik-SB I added a little more info based on my understanding. Thanks Joe!

Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: // convert r4 Libray to R5 Library as we are using r5 liquid engine.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: @adongare Thanks Ashok. Changed.

**Review by adongare — APPROVED** [review]

**Review by jkotanchik-SB — APPROVED** [review]
